libp2p/protocols/pubsub/gossipsub.nim

```diff
@@ -68,6 +165,103 @@ when defined(libp2p_expensive_metrics):
     "gossipsub peers per topic in gossipsub",
     labels = ["topic"])
 
+proc init*(_: type[GossipSubParams]): GossipSubParams =
```

Comment:
nitpic, I think the style we prefer here is:

```suggestion
proc init*(G: type[GossipSubParams]): G =
  G(...)
```

Reply:
Curious I always used my style thru the codebase without noticing the other variation, I count it 13 times. 😄

Reply:
I see other variations like `typedesc[T]` or even (the best !) `xx: type T`

Reply:
Maybe we should open an issue about code refactor and stabilize some styles

Reply:
```
func init(T: type Xxx, ...): T =
  T(
    field: b
  )
```
is the style most nbc code uses (ie `T` consistently for the type, avoiding repeating it

Reply:
Fits also what @dryajov is suggesting, minus the omission of parenthesis